### PR TITLE
fix: bump min supported version of python to 3.8

### PR DESCRIPTION
**Pull request by padamstx**
## PR summary
This commit contains these changes:
- bumps the min supported version of python to 3.8
- change to Makefile to allow unit and int tests to be processed by pylint
- minor changes to existing unit/int tests and examples to avoid lint-check warnings

## PR Checklist
Please make sure that your PR fulfills the following requirements:  
- [ ] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## Does this PR introduce a breaking change?    
- [ ] Yes
- [ ] No

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -3,7 +3,7 @@
 # example: "make setup"
 
 PYTHON=python
-LINT_DIRS=ibm_platform_services test examples
+LINT_DIRS=ibm_platform_services test/unit test/integration examples
```
Comment (padamstx) [inline Makefile:6]: pylint expects modules to be specified, so we were never really lint-checking unit or integration tests before :joy:

Comment (pyrooka) [reply Makefile:6]: Wow, nice catch!

**Review by pyrooka — APPROVED** [review]
Looks good!

**Pull request merged**